src/CMakeLists.txt

```diff
@@ -3,28 +3,46 @@
 cmake_minimum_required(VERSION 3.8)
 
 # set compile options
-set(COMMON_C_FLAGS "-Wall -fsigned-char")
-set(COMMON_CXX_FLAGS "-fno-rtti -ffast-math")
+set(COMMON_CXX_FLAGS "-fno-rtti -ffast-math -fexceptions -Wall -fsigned-char")
 
 if(CMAKE_BUILD_TYPE MATCHES RELEASE)
     message(STATUS "Release build detected, adding release flags")
-    set(COMMON_C_FLAGS "${CMAKE_CXX_FLAGS} -O3 -fomit-frame-pointer")
+    set(COMMON_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2 -fomit-frame-pointer -fstack-protector-all -fpic -fpie -D_FORTIFY_SOURCE=2")
+    if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+        set(COMMON_CXX_FLAGS "${ADDITIONAL_BUILD_FLAGS} -D_GLIBCXX_ASSERTIONS")
```

Comment:
That only really helps when building against libstdc++, I doubt libc++ will pick it up.